src/Framework/IRarBuildEngine.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO.Pipes;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// This interface provides necessary functionality from <see cref="Microsoft.Build"/> to RAR as a service functionality.
+    /// </summary>
+    internal interface IRarBuildEngine
```

Comment:
I'm also hesitant to add this interface but prefer to keep it as is rather than rolling it into IBuildEngine8 so that it's easier to remove later.

Reply:
I wasn't really happy with that too (didn't want to touch the engine at all), but I couldn't find a way how to solve that without this.

Reply:
There is one solution to it, but not sure if it is doable. We would have to link NamePipeUtil into Tasks project (with CommunicationUtilities and XMakeAttribute), this is not an issue (at least from functionality point of view), **but** we would also have to bump up the version of Tasks project from netstnadard2.0 to netstandard2.1, which is quite scary

Reply:
Ah, that is scary _in general_ but for us it's totally fine. There's no case where we need to run Tasks on anything < .NET 5.0 (now), so we should be able to move to `netcoreapp2.1` with no problem at all.

Reply:
As discussed in chat with @rainersigwald we will leave it like that for now. In future it may be removed. The TFM will not be changed.